service/storage_service.cc

```diff
@@ -5456,8 +5473,11 @@ future<raft_topology_cmd_result> storage_service::raft_topology_cmd_handler(raft
                 co_await with_scheduling_group(_db.local().get_streaming_scheduling_group(), coroutine::lambda([&] () -> future<> {
                     const auto& rs = _topology_state_machine._topology.find(raft_server.id())->second;
                     auto tstate = _topology_state_machine._topology.tstate;
-                    if (rs.ring.tokens.empty() ||
-                        (tstate != topology::transition_state::write_both_read_old && rs.state != node_state::normal && rs.state != node_state::rebuilding)) {
+                    if (rs.ring.tokens.empty()) {
+                        rtlogger.info("the node got stream_ranges request but it does not own any tokens");
```

Comment:
Can this even happen? IIUC we shouldn't be sending stream_ranges to these nodes in the first place? (In which case perhaps we can make it `on_internal_error`)

Reply:
`if (rs.ring.tokens.empty() ||` was here before I introduced zero-token nodes. So, its purpose was to handle token-owning nodes with no tokens at the moment. I don't know what case exactly it handles. I don't even know if it handles any case at all. Maybe someone added it "just to be safe" and we don't need it.

I could investigate it. One idea is to use `on_internal_error` and see if any test starts failing. But I would go with `on_internal_error` only if we are sure we can.

Reply:
It turned out I bugged this. The topology coordinator sends `stream_ranges` to zero-token node, for example, when removing a token-owning node. A new test uncovered this bug. The solution is `result.status = raft_topology_cmd_result::command_status::success;`.

Another approach could be to ignore zero-token nodes when sending global `stream_ranges` (and other commands), but I'm not sure if implementing it is worth it.

Reply:
> Another approach could be to ignore zero-token nodes when sending global `stream_ranges` (and other commands), but I'm not sure if implementing it is worth it.

That's what we do in the current version of the PR.